docs/resources/server/tags.md

```diff
@@ -1,83 +1,297 @@
+- Otherwise, assume the tag is not specified in Minecraft or NeoForge, and thus you need to create your own tag.
 
-Tags for all registries are automatically sent from the server to any remote clients on login and reload. `Block`, `Item`, `BlockEntityType`, `EntityType`, `Fluid`, and `GameEvent` are special cased as they have `Holder`s allowing for available tags to be accessible through the object itself.
+When creating your own tag, you should ask yourself the following questions:
 
-:::note
-Intrusive `Holder`s may be removed in a future version of Minecraft. If they are, the below methods can be used instead to query the associated `Holder`s.
-:::
+- Does this modify my mod's behavior? If yes, the tag should be in your mod's namespace. (This is common e.g. for my-thing-can-spawn-on-this-block kind of tags.)
+- Would other mods want to use this tag as well? If yes, the tag should be in the `c` namespace. (This is common e.g. for new metals or gems.)
+- Otherwise, use your mod's namespace.
 
-### Referencing Tags
+Naming the tag itself also has some conventions to follow:
 
-There are two methods of creating a tag wrapper:
+- Use the plural form. E.g.: `minecraft:planks`, `c:ingots`.
+- Use folders for multiple objects of the same type, and an overall tag for each folder. E.g.: `c:ingots/iron`, `c:ingots/gold`, and `c:ingots` containing both. (Note: This is a NeoForge convention, Minecraft does not follow this convention for most tags.)
 
-Method                          | For
-:---:                           | :---
-`*Tags#create`                  | `BannerPattern`, `Biome`, `Block`, `CatVariant`, `DamageType`, `EntityType`, `FlatLevelGeneratorPreset`, `Fluid`, `GameEvent`, `Instrument`, `Item`, `PaintingVariant`, `PoiType`, `Structure`, and `WorldPreset` where `*` represents one of these types.
-`TagKey#create`                 | Registries without vanilla tags.
+## Using Tags
 
-Registry objects can check their tags via `Holder#tags`, getting their `Holder` using either `Registry#getHolder` or `Registry#getHolderOrThrow`. Comparing a single tag can be done using `Holder#is`.
+To reference tags in code, you must create a `TagKey<T>`, where `T` is the type of tag (`Block`, `Item`, `EntityType<?>`, etc.), using a [registry key][regkey] and a [resource location][resloc]:
 
-Tag-holding registry objects also contain a method called `#is` in either their registry object or state-aware class to check whether the object belongs to a certain tag.
+```java
+public static final TagKey<Block> MY_TAG = TagKey.create(
+        // The registry key. The type of the registry must match the generic type of the tag.
+        Registries.BLOCK,
+        // The location of the tag. This example will put our tag at data/examplemod/tags/blocks/example_tag.json.
+        ResourceLocation.fromNamespaceAndPath("examplemod", "example_tag")
+);
+```
+
```

Comment:
Might be worth mentioning that using the `TagKey` constructor directly (it's a record, so it can't be hidden by making it private) will cause issues such as tag entry lookups and tag inclusion checks failing.